### PR TITLE
[4.0] Run a pre-check in app bootup to make sure dependencies are installed

### PR DESCRIPTION
**Pull request by mbabker**
Pull Request for Issue #21278

### Summary of Changes

Run a check during the application bootup to make sure that `composer install` and `npm install` are run (well, relatively sane checks at least).

This uses the same approach as the incompatible PHP version page as it relates to language strings, page design, etc.  If you're reviewing and don't like the JavaScript, you don't like your own script because it's a copy of the PHP page logic 😉

### Testing Instructions

Try to run Joomla without the Composer autoloader (file `libraries/vendor/autoload.php` generated by `composer install`) or the `media/vendor` directory (created by `npm install`).  You'll get a friendly error page pointing out you have some extra work to do and a link to the docs page for the extra info.

### TODO

- Try to DRY up the code for these error pages

### Screenshot

screen shot 2018-07-28 at 3 41 25 pm

### CODE REVIEW

File: build/repo_build_incomplete/incomplete.html [hunk 1]
```diff
@@ -0,0 +1,41 @@
...
+  <!-- Set the content of the translated text, IMPORTANT DO NOT REMOVE -->
+  <script>{{jsonContents}}</script>
+</head>
+<!-- Set the minimum PHP version, IMPORTANT DO NOT REMOVE -->
+<body>
+<div class="container">
+  <div class="container-main">
+    <div class="alert-main">
+      <h1 id="headerText">Environment Setup Incomplete</h1>
+      <p><span id="descText1">It looks like you are trying to run Joomla! from our git repository. To do so requires you complete a couple of extra steps first.</span></p>
```
Comment (brianteeman) [inline build/repo_build_incomplete/incomplete.html:19]: run? wouldn't install be better?

Comment (mbabker) [reply build/repo_build_incomplete/incomplete.html:19]: You can argue it either way.  If you're doing a fresh repo clone, the first thing you'd do would be install, and in that context it's absolutely right to say "install".  After the initial clone though, if you're doing `git pull` or jumping around between branches on a local setup (especially between any 3.x branch and 4.0), there's a chance you hit this page just trying to run your local install (just today I did branch jumping around starting from 4.0 which I had in an installed state, switched over to the privacy PR's branch to merge stuff and push, then back to 4.0, and lost `libraries/vendor/autoload.php` along the way).

**Comment by ggppdk** [comment]
I have tested this item :white_check_mark: successfully on 9e8aadb18345212ad360a7467cdd611bcb6f7b8d<br /><br />Works as described

Just i can not do much of code review on this PR, as i do not have enough knowledge for this<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/21287">issues.joomla.org/tracker/joomla-cms/21287</a>.</sub>

File: build/repo_build_incomplete/incomplete.html (hunk 1 above)
Comment (brianteeman) [reply build/repo_build_incomplete/incomplete.html:19]: ok - didnt realise that would happen when switching between branches - so all good for me

**Review by brianteeman — APPROVED** [review]

**Comment by Fedik** [comment]
I have tested this item :white_check_mark: successfully on 9e8aadb18345212ad360a7467cdd611bcb6f7b8d<br /><br />very cool!<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/21287">issues.joomla.org/tracker/joomla-cms/21287</a>.</sub>

**Comment by Quy** [comment]
RTC<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/21287">issues.joomla.org/tracker/joomla-cms/21287</a>.</sub>

**Comment by brianteeman** [comment]
I have tested this item :white_check_mark: successfully on 9e8aadb18345212ad360a7467cdd611bcb6f7b8d<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/21287">issues.joomla.org/tracker/joomla-cms/21287</a>.</sub>

**Comment by wilsonge** [comment]
Conflicts please

**Review by wilsonge — APPROVED** [review]

**Comment by mbabker** [comment]
Rebased.

**Comment by laoneo** [comment]
Thanks

**Pull request merged**